features/unattached_commands.feature

```diff
@@ -529,6 +529,46 @@ Feature: Command behaviour when unattached
           | impish  |
           | jammy   |
 
+    @series.all
+    @uses.config.machine_type.lxd.container
+    Scenario Outline: Unattached attach with json output
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I verify that running attach `as non-root` with json response exits `1`
+        Then I will see the following on stdout:
+            """
+            {"_schema_version": "0.1", "errors": [{"message": "This command must be run as root (try using sudo).", "service": null, "type": "system"}], "failed_services": [], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
+            """
+        When I verify that running `ua attach invalid-token --format json` `with sudo` exits `1`
+        Then I will see the following on stdout:
+            """
+            {"_schema_version": "0.1", "errors": [{"message": "Invalid token. See https://ubuntu.com/advantage", "service": null, "type": "system"}], "failed_services": [], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
+            """
+        When I verify that running attach `with sudo` with json response exits `0`
```

Comment:
If we want attach to work here, I think we could move this whole test to `attach_validtoken.feature.`

Reply:
I would avoid splitting the tests like this because we would need to spin a different container just to test that. Or do you mean changing this bit into an existing test on `attach_validtoken` ?

Reply:
Well this one is spinning a container for it, right? I just think it should live somewhere else.

Reply:
True, then we would be spinning two containers if we move it, right ?

Reply:
No, sorry, I don't understand why.
But if you think it is best to leave this here, it's all right.

Reply:
I think I am the one that missed your point. I thought you meant to partially move this test to `attach_validtoken` (that's why I commented that we would then need two containers to test the whole feature), but you mean to move the entire test which makes sense, I will perform this change